### PR TITLE
ST: Use cmd client for deploy in upgrade tests

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Refactoring

### Description

We should use cmd client for each CR creation during the upgrade because at some point, we will need to deal with different api versions and it cannot be handled by fabric8 client easily.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 1]
```diff
@@ -322,4 +324,13 @@ public static String changeDeploymentNamespace(File deploymentFile, String names
...
     }
+
+    public static void waitForResourceReadiness(String resourceType, String resourceName) {
+        LOGGER.info("Waiting for " + resourceType + "/" + resourceName + " readiness");
+
+        TestUtils.waitFor("resource " + resourceType + "/" + resourceName + " readiness",
+                Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_CMD_CLIENT_TIMEOUT,
+            () -> ResourceManager.cmdKubeClient().getResourceReadiness(resourceType, resourceName));
+        LOGGER.info("Resource " + resourceType + "/" + resourceName + " is ready");
+    }
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java:335]: Wouldn't be better to add this to `ResourceManager`? Both options - have it in `StUtils` or `ResourceManager` - makes sense to me. So it's up to you

**Review by im-konge — APPROVED** [review]
Just one suggestion. LGTM

**Review by scholzj — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java:335]: Make sense to move it to ResourceManager

**Pull request merged**